src/services/codefixes/inferFromUsage.ts

```diff
@@ -980,7 +980,12 @@ namespace ts.codefix {
             return types;
         }
 
-        function inferStructuralType(usage: Usage) {
+        function inferLiteralOrStructuralType(usage: Usage) {
+            const matchingBuiltins = builtinPrimitives.filter(builtin => allPropertiesAreAssignableToUsage(builtin, usage));

```

Comment:
This new code is redundant with the code in `inferNamedTypesFromProperties`, which is called a couple of lines below `inferLiteralOrStructuralType`. Why doesn't the code there work correctly? I think fixing that code is the better fix.

Reply:
Later: see other comments for what I investigated. I'm pretty sure the problem is mostly in how often we infer `void`.